### DISCUSSION TITLE
Changes to Internal API/Service Domains, Upcoming Adjustment to Link to Media field

### DISCUSSION THREAD

**Discussion by kilianweisl** [prismicio/php-kit#205] [Q&A] [answered]
Hello!

I recently received the following E-Mail from Prismic.
Since this repository is community-managed, I wanted to ask if these changes are already handled by the package, especially regarding the kind property?

--- 

Dear Prismic User,

We hope this message finds you well.

We are writing to inform you about two important upcoming changes that will take effect on November 15, 2024. Please review the following information carefully to ensure a smooth transition.

1. Changes to Internal API/Service Domains
To improve the scalability, performance, and security of our internal APIs and services, we are updating our architecture.

Starting November 15, 2024, we will start consolidating our internal APIs and services under one unique subdomain: internal.prismic.io.
 
What this means for you:
For most users no action is needed but If your project is behind a secure network or VPN with specific domain restrictions, you may need to update your network or VPN policy to allow access to the new domain. If this applies to you, we recommend whitelisting the following domain name: *.internal.prismic.io/*
 
2. Upcoming Adjustment to Link to Media field

Effective November 15, 2024, in our Public Content API, we will adjust the kind property of the field Links to Media. 

Currently, the Public Content API can return inconsistent kind property (for the same type of Media you could receive image, audio, document, video, file or all as a value)

Starting November 15, we're fixing this bug, Hence, the API will return only image or file for the kind property
 
What this means for you:

If you're not using the kind property in your code or if your code solely relies on our SDKs (PrismicNextLink or PrismicLink), no action is needed. 

If your code references any of the unsupported values (audio, document, video, or all), you will need to update it to support the value file. After November 15, 2024, the unsupported values will no longer be returned

Need Assistance?
If you're unsure whether these changes impact you, or if you have any questions, please feel free to reach out to our support team or contact your Customer Success Manager. We're ready to assist you to ensure a smooth transition.

Thank you for your attention and for being a valued member of the Prismic community.

Best regards,

Prismic Support Team

**Comment by c0nst4ntin** [comment]
@kilianweisl Thanks for the questions! I took care of the PHP 8 update and since then try to keep an eye on the package.

I also recently received this message and still need to look into this a bit deeper.

@angeloashmore Can you provide me with some more detailed information on these changes? Is there any way I can test the api with these changes?

**Reply by c0nst4ntin** [reply]
@angeloashmore Did you manage to check this? Please let me know if there are any changes needed or how we should proceed.

**Reply by angeloashmore** [reply, marked as answer]
Hi @c0nst4ntin, apologies for the late reply. The API changes should have no effect on the PHP SDK.

**Changes to Internal API/Service Domains**: This primarily affects the Prismic app (i.e. the Page Builder). The Document API URLs will not change.

**Upcoming Adjustment to Link to Media field**: This change will affect API's `kind` property for link fields. The SDK does not read the `kind` value, thus it should not be affected.

**Reply by c0nst4ntin** [reply]
@angeloashmore Thanks for clarifying! If anybody should still experience any troubles please create an issue so we can resolve this as soon as possible!

**Comment by kilianweisl** [comment]
Thanks @c0nst4ntin and @angeloashmore for your effort!
